### PR TITLE
[WFLY-3045] : Expose JAXRS deployments via the management API

### PR DESCRIPTION
**Pull request by ehsavoie**
Displaying JAXRS resources in deployments, showing paths and  methods.
Adding mappings to the deployment infos of servlets.

Jira: https://issues.jboss.org/browse/WFLY-3045

### CODE REVIEW

**Comment by ehsavoie** [comment]
retest this please

File: jaxrs/src/main/java/org/jboss/as/jaxrs/JaxrsDeploymentDefinition.java [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+                        @Override
+                        public void execute(final OperationContext context, final ModelNode operation) throws OperationFailedException {
+                            if (registry != null) {
+                                final ModelNode response = new ModelNode();
+                                for (Map.Entry<String, List<ResourceInvoker>> resource : registry.getBounded().entrySet()) {
+                                    handle(response, contextPath, servletMappings, resource.getKey(), resource.getValue());
+                                }
+                                context.getResult().set(response);
+                            }
+                            context.stepCompleted();
```
Comment (ctomc) [inline jaxrs/src/main/java/org/jboss/as/jaxrs/JaxrsDeploymentDefinition.java]: not needed anymore

File: jaxrs/src/main/java/org/jboss/as/jaxrs/JaxrsDeploymentDefinition.java [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+                        }
+                    }, OperationContext.Stage.RUNTIME);
+                    context.stepCompleted();
```
Comment (ctomc) [inline jaxrs/src/main/java/org/jboss/as/jaxrs/JaxrsDeploymentDefinition.java]: not needed anymore

File: feature-pack/src/main/resources/modules/system/layers/base/org/jboss/as/jaxrs/main/module.xml [hunk 3]
```diff
@@ -39,12 +39,16 @@
...
         <module name="javax.ws.rs.api"/>
+        <module name="javax.servlet.api"/>
         <module name="org.jboss.as.controller"/>
         <module name="org.jboss.as.ee"/>
         <module name="org.jboss.as.ejb3"/>
         <module name="org.wildfly.security.manager"/>
         <module name="org.jboss.as.server"/>
         <module name="org.jboss.as.web-common"/>
+        <module name="org.wildfly.extension.undertow"/>
+        <module name="io.undertow.core"/>
```
Comment (ctomc) [inline feature-pack/src/main/resources/modules/system/layers/base/org/jboss/as/jaxrs/main/module.xml]: really needed? the servlet one sure, but not sure for undertow core

**Comment by ehsavoie** [comment]
retest this please

**Pull request merged**